RenderSystems/GLSupport/src/EGL/Wayland/OgreWaylandEGLWindow.cpp

```diff
@@ -0,0 +1,297 @@
+// This file is part of the OGRE project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at https://www.ogre3d.org/licensing.
+// SPDX-License-Identifier: MIT
+
+#include "OgreException.h"
+#include "OgreLogManager.h"
+#include "OgreRoot.h"
+#include "OgreStringConverter.h"
+#include "OgreViewport.h"
+
+#include "OgreWaylandEGLSupport.h"
+#include "OgreWaylandEGLWindow.h"
+
+#include <thread>
+#include <chrono>
+
+namespace Ogre
+{
+
+WaylandEGLWindow::WaylandEGLWindow(WaylandEGLSupport* glsupport) : EGLWindow(glsupport)
+{
+    mGLSupport = glsupport;
+    mNativeDisplay = nullptr; //glsupport->getNativeDisplay();
+}
+
+WaylandEGLWindow::~WaylandEGLWindow()
+{
+    //mNativeDisplay = mGLSupport->getNativeDisplay();
+
+    if (mWindow && mIsTopLevel)
+    {
+        if (!mIsExternal)
+            wl_egl_window_destroy(mWindow);
+    }
+    mWindow = nullptr;
+}
+
+void WaylandEGLWindow::initNativeCreatedWindow(const NameValuePairList* miscParams)
+{
+    //mNativeDisplay = mGLSupport->getNativeDisplay();
+
+    if (miscParams)
+    {
+        NameValuePairList::const_iterator opt;
+        NameValuePairList::const_iterator end = miscParams->end();
+
+        if ((opt = miscParams->find("externalWlDisplay")) != end)
+        {
+          StringVector tokens = StringUtil::split(opt->second, " :");
+
+          auto tmp = (wl_display*)StringConverter::parseSizeT(tokens[0]);
+          mNativeDisplay = tmp;
+          mGLSupport->setNativeDisplay(mNativeDisplay);
+
+        }
+        if ((opt = miscParams->find("externalSurface")) != end)
+        {
+            StringVector tokens = StringUtil::split(opt->second, " :");
+
+            mGLSupport->mWlSurface = (wl_surface*)StringConverter::parseSizeT(tokens[0]);
+        }
+    }
+    mNativeDisplay = mGLSupport->getNativeDisplay();
+}
+
+void WaylandEGLWindow::createNativeWindow(uint& width, uint& height, String& title)
+{
+    mEglDisplay = mGLSupport->getGLDisplay();
+
+    mGLSupport->mWlRegion = wl_compositor_create_region(mGLSupport->mWlCompositor);
+    wl_region_add(mGLSupport->mWlRegion, 0, 0, width, height);
+    wl_surface_set_opaque_region(mGLSupport->mWlSurface, mGLSupport->mWlRegion);
```

Comment:
do we have to do this ourselves? I would assume the surface is sufficiently configured by SDL/ Qt..

Reply:
This is indeed not needed. By not doing this the `alpha channel` artifacts are removed too.